client/blocks/qr-code-login/index.jsx

```diff
@@ -0,0 +1,198 @@
+import { getTracksAnonymousUserId } from '@automattic/calypso-analytics';
+import { Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import QRCode from 'qrcode.react';
+import { useEffect, useState, useRef } from 'react';
+import qrCenter from 'calypso/assets/images/qr-login/wp.png';
+
+import './style.scss';
+
+const AUTH_PULL_INTERVAL = 5000; // 5 seconds
+const LOCALE_STORAGE_KEY = 'qr-login-token';
+
+const isStillValidToken = ( tokenData ) => {
+	if ( ! tokenData ) {
+		return false;
+	}
+	const { expires } = tokenData;
+	if ( ! expires ) {
+		return false;
+	}
+	return expires > Date.now() / 1000;
+};
+
+const setLocalTokenData = ( data ) => {
+	try {
+		window.localStorage.setItem( LOCALE_STORAGE_KEY, JSON.stringify( data ) );
```

Comment:
I wonder if we can use the `lib/browser-storage` library for storing the token data. It will handle (de)serialization and will also attempt to use IndexedDB before falling back to local storage. It also has some tests 😉

Reply:
I updated the code to now use this library.